src/fable-library/Array.fs

```diff
@@ -877,13 +877,23 @@ let tail (array: 'T[]) =
 
     skipImpl array 1
 
-let item index (array: _[]) = array.[index]
+let item (index: int) (array: 'T[]) : 'T =
+    if index < 0 || index >= array.Length then
+        invalidArg "index" "Index was outside the bounds of the array."
+    else
+        // We need to emit JavaScript directly here
+        // otherwise we will call `item` function recursively
+        // and it will cause a stack overflow
+        emitJsExpr (index, array) "$1[$0]"
```

Comment:
@MangelMaxime Perhaps move this to `Native.fs`.

Reply:
I didn't know about this file.

I will look to add an helper in that module